gammapy/modeling/parameter.py

```diff
@@ -264,10 +258,42 @@ def max(self):
     @max.setter
     def max(self, val):
         """`~astropy.table.Table` has masked values for NaN. Replacing with NaN."""
-        if isinstance(val, np.ma.core.MaskedConstant):
-            self._max = np.nan
+        self._max = self._set_min_max(val)
+
+    def _set_min_max(self, value):
+        """Logics for min and max setter."""
+        if isinstance(value, np.ma.core.MaskedConstant) or (value is None):
+            return np.nan
+        elif isinstance(value, u.Quantity):
+            return float(value.to(self.unit).value)
+        elif isinstance(value, str):
+            return float(u.Quantity(value).to(self.unit).value)
         else:
-            self._max = float(val)
+            return float(value)
+
+    def set_lim(self, min=None, max=None):
+        """
+        Set the min and/or max value for the parameter.
+
+        Parameters
+        ----------
+        min, max: float, `~astropy.units.Quantity`, str or tuple of float/`~astropy.units.Quantity`/str
+            Minimum and Maximum value to assign to the parameter `min` and `max`. If a tuple is
+            passed and `max` is set to None, the tuple will be decomposed into min and max value.
+            Default is None, which set `min` and `max` to `np.nan`.
+        """
+        if max is None and np.iterable(min):
```

Comment:
Here you should raise an error if min is a tuple and max is also present. Otherwise it will fail with a confusing message later on

Reply:
Agree !
By the way, I did this because this method is largely inspired by `plt.x/ylim`, `Axes.set_x/ylim`. 
In a way I like that you can have tuples for min and max values and directly pass them to this method. But it makes the API a bit weird, so if people have strong opinion against it, we could remove this possibility. 
Otherwise, we could also show the call signature in the docstring, as it is done in the [x/ylim documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.xlim.html).

Reply:
I agree with you about the convenience, and also about the API being a bit weird to have an iterable for min.
Maybe the user can either set `lims = ()` or min and max separately?